test/lint/lint-python-dead-code.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+Find dead Python code.
+"""
+
+from subprocess import check_output, STDOUT, CalledProcessError
+
+FILES_ARGS = ['git', 'ls-files', '--', '*.py']
+
+
+def check_vulture_install():
+    try:
+        check_output(["vulture", "--version"])
+    except FileNotFoundError:
+        print("Skipping Python dead code linting since vulture is not installed. Install by running \"pip3 install vulture\"")
+        exit(0)
+
+
+def main():
+    check_vulture_install()
+
+    files = check_output(FILES_ARGS).decode("utf-8").splitlines()
+    # --min-confidence 100 will only report code that is guaranteed to be unused within the analyzed files.
+    # Any value below 100 introduces the risk of false positives, which would create an unacceptable maintenance burden.
+    vulture_args = ['vulture', '--min-confidence=100'] + files
+
+    try:
+        check_output(vulture_args, stderr=STDOUT)
```

Comment:
Will this print the error? I'd guess no

Reply:
Hm, for me it does. It doesn't for you?

(faked to get an alert)
```
$ test/lint/lint-python-dead-code.py
/Users/FJ/projects/clones/bitcoin/test/lint/lint-python-dead-code.py:26: unsatisfiable 'if' condition (100% confidence)
Python dead code detection found some issues
```

Reply:
Thanks for testing and confirming. I just assumed that redirecting stderr to stdout and then capturing stdout, which is done by `check_output`, which discards the return value here, would also drop the stderr.

This is also what the documentation says:

https://docs.python.org/3/library/subprocess.html#subprocess.check_output

> To also capture standard error in the result, use `stderr=subprocess.STDOUT`


Not sure what to do now. Are you trying to trick me into installing `vulture`? :thinking:

Reply:
Oh, sorry. I missed the `e.output.decode` two lines below. :man_facepalming: